Tests/FluentAssertions.Specs/Execution/TestFrameworkProviderTests.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using FluentAssertions.Common;
+using FluentAssertions.Execution;
+using Xunit;
+
+namespace FluentAssertions.Specs.Execution;
+
+public class TestFrameworkProviderTests
+{
+    [Fact]
+    public void When_running_xunit_test_implicitly_it_should_be_detected()
+    {
+        var configuration = new Configuration(new TestConfigurationStore());
+        var result = TestFrameworkProvider.DetectFramework(configuration);
+
+        result.IsAvailable.Should().BeTrue();
+        result.Should().BeOfType<XUnit2TestFramework>();
+    }
+
+    [Fact]
+    public void When_running_xunit_test_explicitly_it_should_be_detected()
+    {
+        var configurationStore = new TestConfigurationStore { Data = { [Configuration.TestFrameworkConfigurationKey] = "xunit2" } };
+        var configuration = new Configuration(configurationStore);
+        var result = TestFrameworkProvider.DetectFramework(configuration);
+
+        result.IsAvailable.Should().BeTrue();
+        result.Should().BeOfType<XUnit2TestFramework>();
+    }
+
+    [Fact]
+    public void When_running_test_with_unknown_test_framework_it_should_throw()
+    {
+        var configurationStore = new TestConfigurationStore { Data = { [Configuration.TestFrameworkConfigurationKey] = "foo" } };
+        var configuration = new Configuration(configurationStore);
+
+        this.Invoking(_ => TestFrameworkProvider.AttemptToDetectUsingAppSetting(configuration))
```

Comment:
If you add
```cs
using static FluentAssertions.FluentActions;
```

You can write this as
```cs
Invoking(() => TestFrameworkProvider.AttemptToDetectUsingAppSetting(configuration))
```

In the end it is functionally equivalent to the existing, but to keep the majority of the test suite a prime example of how to write unit tests using Fluent Assertions.

Reply:
done